File: src/main/java/com/epicsquid/whoosh/containers/TransporterMenu.java
package com.epicsquid.whoosh.containers;

import cofh.core.inventory.container.ContainerCoFH;
import net.minecraft.network.FriendlyByteBuf;
import net.minecraft.world.entity.player.Inventory;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.inventory.MenuType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class TransporterMenu extends ContainerCoFH {

	public TransporterMenu(@Nullable MenuType<?> type, int id, Inventory inventory, FriendlyByteBuf buf) {
		super(type, id, inventory, inventory.player);
	}

	public TransporterMenu(@Nullable MenuType<?> type, int id, Inventory inventory, Player player) {
		super(type, id, inventory, player);
	}

	@Override
	protected int getMergeableSlotCount() {
		return 0;
	}

	@Override
	public boolean stillValid(@NotNull Player player) {
		return true;
	}
}

File: src/main/java/dev/epicsquid/whoosh/containers/TransporterMenu.kt
package dev.epicsquid.whoosh.containers

import cofh.core.inventory.container.ContainerCoFH
import dev.epicsquid.whoosh.registery.WhooshLang
import dev.epicsquid.whoosh.registery.WhooshMenuTypes
import net.minecraft.network.chat.Component
import net.minecraft.world.MenuProvider
import net.minecraft.world.entity.player.Inventory
import net.minecraft.world.entity.player.Player
import net.minecraft.world.inventory.AbstractContainerMenu
import net.minecraft.world.inventory.MenuType

//
//import cofh.core.inventory.container.ContainerCoFH
//import dev.epicsquid.whoosh.registery.WhooshLang
//import dev.epicsquid.whoosh.registery.WhooshMenuTypes
//import net.minecraft.network.chat.Component
//import net.minecraft.world.MenuProvider
//import net.minecraft.world.entity.player.Inventory
//import net.minecraft.world.entity.player.Player
//import net.minecraft.world.inventory.AbstractContainerMenu
//import net.minecraft.world.inventory.MenuType
//
class TransporterMenu(type: MenuType<TransporterMenu>, id: Int, inventory: Inventory?, player: Player?) :
	ContainerCoFH(type, id, inventory, player), MenuProvider {


	//
//
//
	override fun stillValid(pPlayer: Player): Boolean {
		return true
	}
//
	override fun getMergeableSlotCount(): Int {
		return 0
	}

	override fun createMenu(pContainerId: Int, pPlayerInventory: Inventory, pPlayer: Player): AbstractContainerMenu? {
		return TransporterMenu(WhooshMenuTypes.TRANSPORTER, pContainerId, pPlayerInventory, pPlayer)

	}

	override fun getDisplayName(): Component {
		return WhooshLang.TRANSPORTER


}
